### DISCUSSION TITLE
What is the value of langserve? Is it for the personal end, not the server end?

### DISCUSSION THREAD

**Discussion by cgafool** [langchain-ai/langserve#358] [Q&A] [answered]
The difference between langserve and TGI, vLLM, or why do I need langserve when I already have vLLM?

**Comment by Vinno97** [comment, marked as answer]
LangServe hosts LangChain apps built on top of LLMs. vLLM hosts a raw LLM

**Comment by eyurtsev** [comment]
Don't use LangServe for serving raw LLMs -- it's not designed for that use case and will not manage the hardware resources appropriately. It's specifically for serving LangChain apps built on tops of LLMs.
